content/docs/guides/parallel-execution.md

```diff
@@ -0,0 +1,380 @@
+      | scen_out_row_num       |
+      | Scenario Outline Row 1 |
+      | Scenario Outline Row 2 |
+```
+
+- Add the **step definition class** to the `parallel` package (same name as folder above for automatic pickup by runner) in `src/test/java` folder.
+{{% /block %}}
+
+{{% block "java" %}}
+
+```java
+package parallel;
+
+import cucumber.api.java.BeforeStep;
+import cucumber.api.java.en.Given;
+
+public class Stepdefs {
+
+	@Given("Step from {string} in {string} feature file")
+	public void step(String scenario, String file) {
+		System.out.format("Thread ID - %2d - %s from %s feature file.\n", Thread.currentThread().getId(), scenario,file);
+	}
+}
+```
+{{% /block %}}
+
+{{% block "kotlin" %}}
+
+```kotlin
+package parallel
+
+import cucumber.api.java8.En
+
+class Stepdefs : En {
+		
+		Given("Step from {string} in {string} feature file") { scenario: String , file: String ->
+			println("Thread ID - ${Thread.currentThread().id} - $scenario from $file feature file")
+       }
+	}
+}
+```
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+- Add a cucumber **runner** using the `RunWith` annotation in the `parallel` package (same name as step definition package) in the `src/test/java` folder.
+{{% /block %}}
+
+{{% block "java" %}}
+
+```java
+package parallel;
+
+import cucumber.api.junit.Cucumber;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+public class RunCucumberTest {
+}
+```
+{{% /block %}}
+
+{{% block "kotlin" %}}
+
+```kotlin
+package parallel
+
+import cucumber.api.junit.Cucumber
+import org.junit.runner.RunWith
+
+@RunWith(Cucumber::class)
+class RunCucumberTest {
+}
+```
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+- Add the **Surefire plugin configuration** to the `build` section to the `POM`.
+
+```shell
+<plugin>
+	<groupId>org.apache.maven.plugins</groupId>
+	<artifactId>maven-surefire-plugin</artifactId>
+	<version>2.22.0</version>
+	<configuration>
+		<parallel>methods</parallel>
+		<useUnlimitedThreads>true</useUnlimitedThreads>
+	</configuration>
+</plugin>
+```
+
+- Use the Maven `install` or a suitable command to **execute** the `POM`. This should run in parallel threaded mode. You should see a result similar to below. It is important to note that **both the scenarios in the file (`scenarios.feature`) are executed by thread with ID 14**. Similarly **both the rows of the scenario outline in the file (`scenario-outlines.feature`) are executed by thread with ID 13**.
+
+```shell
+Thread ID - 13 - Scenario Outline Row 1 from scenario-outlines feature file.
+Thread ID - 13 - Scenario Outline Row 2 from scenario-outlines feature file.
+Thread ID - 14 - Scenario 1 from scenarios feature file.
+Thread ID - 14 - Scenario 2 from scenarios feature file.
+```
+
+- To execute using a Maven **Failsafe plugin include the below configuration** in the `build` section to the `POM`. Rename the runner class to `RunCucumberIT`.  You can find further details [here](/docs/community/not-cucumber/#maven-execution-plugin).
+

```

Comment:
```suggestion
{{% block "kotlin" %}}
For Failsafe to find your step definitions, make sure they are in src/test/**java**.
{{% /block %}}
```

Reply:
Alternatively, it might be possible to configure Failsafe to use `src/test/kotlin`, but I don't know how.